radixdlt-java/src/main/java/com/radixdlt/client/core/network/websocket/WebSocketClient.java

```diff
@@ -0,0 +1,140 @@
+package com.radixdlt.client.core.network.websocket;
+
+import com.radixdlt.client.core.network.jsonrpc.PersistentChannel;
+import io.reactivex.Observable;
+import io.reactivex.subjects.BehaviorSubject;
+import io.reactivex.subjects.PublishSubject;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import okhttp3.Response;
+import okhttp3.WebSocket;
+import okhttp3.WebSocketListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Manages the state for a single websocket.
+ */
+public class WebSocketClient implements PersistentChannel {
+	private static final Logger LOGGER = LoggerFactory.getLogger(WebSocketClient.class);
+	private final Object lock = new Object();
```

Comment:
I think it should be LOCK since it's effectively a constant. Martin pointed this out in an earlier PR and it should apply here as well

Reply:
I think it should be `LOCK` only if it's static no? The lock in this case is not shared between instances.